### ISSUE TITLE
Polygamma function for n>=2

### ISSUE THREAD

**Issue by pandeydeep9** [pytorch/pytorch#40980]
## 🚀 Feature

In pytorch, polygamma function is not implemented for n>=2. I need to use higher-order polygamma functions in loss functions.
## Motivation
I am running experiments where I need to experiment with higher order derivatives of digamma function. I can't find a solution for how to do this.

## Pitch
I want to be able to have output for polygamma at least upto n =3.

## Alternatives
I have not found any good alternative for this in pytorch.

**Comment by albanD** [comment]
Hi,

Thanks for the report. We would be happy to accept a PR adding this feature!

**Comment by ita9naiwa** [comment]
Hi @pandeydeep9, and @albanD 

I needed this feature, so I searched and came around here!

I found some implementation references:
1. Scipy https://github.com/scipy/scipy/blob/v1.5.2/scipy/special/_basic.py#L1077-L1117

2. Tensorflow https://github.com/tensorflow/tensorflow/pull/1834

This wouldn't be extremely tough. if @pandeydeep9 didn't embark already, I can go on it...!

**Comment by pandeydeep9** [comment]
Hi @ita9naiwa I have not started working on it. If you want, you can go ahead. Let me know once you are done.

**Comment by ita9naiwa** [comment]
@pandeydeep9  #42499 

I will finish GPU version of it in two or three days, because I'm not able to do tests on GPU machines now
